[Master][HTTP] No bean could be found in the registry for: syndesisHeaderStrategy

## This is a...

<pre><code>
[ ] Feature request
[X] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description

Any integration with HTTP endpoint fails with: 

```
org.apache.camel.RuntimeCamelException: org.apache.camel.FailedToCreateRouteException: Failed to create route -Ld7x2rredgAbVyxhcJc at: >>> To[http4-0-0] <<< in route: Route(-Ld7x2rredgAbVyxhcJc)[[From[timer:integration?period=3... because of Failed to resolve endpoint: http4-0-0 due to: Failed to resolve endpoint: http4://http-svc:8080/?headerFilterStrategy=syndesisHeaderStrategy&httpClient.redirectsEnabled=true&httpMethod=GET due to: No bean could be found in the registry for: syndesisHeaderStrategy of type: org.apache.camel.spi.HeaderFilterStrategy
	at org.apache.camel.util.ObjectHelper.wrapRuntimeCamelException(ObjectHelper.java:1830) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.spring.SpringCamelContext.start(SpringCamelContext.java:136) ~[camel-spring-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.spring.SpringCamelContext.onApplicationEvent(SpringCamelContext.java:174) ~[camel-spring-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:393) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:347) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:883) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:144) ~[spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) ~[spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at io.syndesis.example.Application.main(Application.java:13) [classes!/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_191]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_191]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_191]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_191]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:595) [project-0.1-SNAPSHOT.jar:na]
Caused by: org.apache.camel.FailedToCreateRouteException: Failed to create route -Ld7x2rredgAbVyxhcJc at: >>> To[http4-0-0] <<< in route: Route(-Ld7x2rredgAbVyxhcJc)[[From[timer:integration?period=3... because of Failed to resolve endpoint: http4-0-0 due to: Failed to resolve endpoint: http4://http-svc:8080/?headerFilterStrategy=syndesisHeaderStrategy&httpClient.redirectsEnabled=true&httpMethod=GET due to: No bean could be found in the registry for: syndesisHeaderStrategy of type: org.apache.camel.spi.HeaderFilterStrategy
	at org.apache.camel.model.RouteDefinition.addRoutes(RouteDefinition.java:1303) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.model.RouteDefinition.addRoutes(RouteDefinition.java:204) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext.startRoute(DefaultCamelContext.java:1143) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:3729) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext.doStartCamel(DefaultCamelContext.java:3443) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext.access$000(DefaultCamelContext.java:209) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext$2.call(DefaultCamelContext.java:3251) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext$2.call(DefaultCamelContext.java:3247) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext.doWithDefinedClassLoader(DefaultCamelContext.java:3270) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext.doStart(DefaultCamelContext.java:3247) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:3163) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.spring.SpringCamelContext.start(SpringCamelContext.java:133) ~[camel-spring-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	... 24 common frames omitted
Caused by: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint: http4-0-0 due to: Failed to resolve endpoint: http4://http-svc:8080/?headerFilterStrategy=syndesisHeaderStrategy&httpClient.redirectsEnabled=true&httpMethod=GET due to: No bean could be found in the registry for: syndesisHeaderStrategy of type: org.apache.camel.spi.HeaderFilterStrategy
	at org.apache.camel.impl.DefaultCamelContext.getEndpoint(DefaultCamelContext.java:758) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.util.CamelContextHelper.getMandatoryEndpoint(CamelContextHelper.java:80) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.model.RouteDefinition.resolveEndpoint(RouteDefinition.java:219) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultRouteContext.resolveEndpoint(DefaultRouteContext.java:115) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultRouteContext.resolveEndpoint(DefaultRouteContext.java:121) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.model.SendDefinition.resolveEndpoint(SendDefinition.java:62) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.model.SendDefinition.createProcessor(SendDefinition.java:56) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.model.ProcessorDefinition.makeProcessorImpl(ProcessorDefinition.java:562) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.model.ProcessorDefinition.makeProcessor(ProcessorDefinition.java:523) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.model.ProcessorDefinition.addRoutes(ProcessorDefinition.java:239) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.model.RouteDefinition.addRoutes(RouteDefinition.java:1300) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	... 36 common frames omitted
Caused by: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint: http4://http-svc:8080/?headerFilterStrategy=syndesisHeaderStrategy&httpClient.redirectsEnabled=true&httpMethod=GET due to: No bean could be found in the registry for: syndesisHeaderStrategy of type: org.apache.camel.spi.HeaderFilterStrategy
	at org.apache.camel.impl.DefaultCamelContext.getEndpoint(DefaultCamelContext.java:758) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at io.syndesis.connector.http.HttpConnectorFactories$Http4$1.createDelegateEndpoint(HttpConnectorFactories.java:68) ~[connector-http-1.7-SNAPSHOT.jar!/:1.7-SNAPSHOT]
	at io.syndesis.integration.component.proxy.ComponentProxyComponent.createEndpoint(ComponentProxyComponent.java:133) ~[integration-component-proxy-1.7-SNAPSHOT.jar!/:1.7-SNAPSHOT]
	at org.apache.camel.impl.DefaultComponent.createEndpoint(DefaultComponent.java:130) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext.getEndpoint(DefaultCamelContext.java:743) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	... 46 common frames omitted
Caused by: org.apache.camel.NoSuchBeanException: No bean could be found in the registry for: syndesisHeaderStrategy of type: org.apache.camel.spi.HeaderFilterStrategy
	at org.apache.camel.util.CamelContextHelper.mandatoryLookupAndConvert(CamelContextHelper.java:201) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.util.EndpointHelper.resolveReferenceParameter(EndpointHelper.java:324) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.util.EndpointHelper.resolveReferenceParameter(EndpointHelper.java:306) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultComponent.resolveAndRemoveReferenceParameter(DefaultComponent.java:429) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultComponent.resolveAndRemoveReferenceParameter(DefaultComponent.java:408) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.component.http4.HttpComponent.createEndpoint(HttpComponent.java:234) ~[camel-http4-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultComponent.createEndpoint(DefaultComponent.java:130) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	at org.apache.camel.impl.DefaultCamelContext.getEndpoint(DefaultCamelContext.java:711) ~[camel-core-2.21.0.fuse-740001.jar!/:2.21.0.fuse-740001]
	... 50 common frames omitted
```

--- a/app/integration/runtime-springboot/pom.xml
+++ b/app/integration/runtime-springboot/pom.xml
@@ -76,6 +76,10 @@
       <groupId>org.apache.camel</groupId>
       <artifactId>camel-spring</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.camel</groupId>
+      <artifactId>camel-servlet</artifactId>
+    </dependency>
 
     <!-- spring -->
     <dependency>
@@ -218,10 +222,6 @@
       <artifactId>JUnitParams</artifactId>
       <version>1.1.1</version>
       <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.camel</groupId>
-      <artifactId>camel-servlet</artifactId>
     </dependency>
 
   </dependencies>

--- a/app/integration/runtime-springboot/src/main/java/io/syndesis/integration/runtime/sb/SyndesisHttpConfiguration.java
+++ b/app/integration/runtime-springboot/src/main/java/io/syndesis/integration/runtime/sb/SyndesisHttpConfiguration.java
@@ -21,6 +21,7 @@ import org.springframework.boot.autoconfigure.AutoConfigureBefore;
 import org.springframework.boot.autoconfigure.condition.AnyNestedCondition;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnResource;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Conditional;
 import org.springframework.context.annotation.Configuration;
@@ -48,15 +49,23 @@ public class SyndesisHttpConfiguration {
         /**
          * Checks if we're using API connector-http.
          */
-        @ConditionalOnClass(name = "io.syndesis.connector.http.HttpVerifier")
+        @ConditionalOnResource(resources = "classpath:META-INF/syndesis/connector/http4.json")
         static class HttpUsed {
+            // auto configuration test
+        }
+
+        /**
+         * Checks if we're using API connector-https.
+         */
+        @ConditionalOnResource(resources = "classpath:META-INF/syndesis/connector/https4.json")
+        static class HttpsUsed {
             // auto configuration test
         }
 
         /**
          * Checks if we're using API connector-rest-swagger.
          */
-        @ConditionalOnClass(name = "io.syndesis.connector.rest.swagger.SwaggerConnectorComponent")
+        @ConditionalOnResource(resources = "classpath:META-INF/syndesis/connector/rest-swagger.json")
         static class SyndesisApiClientUsed {
             // auto configuration test
         }
@@ -64,7 +73,7 @@ public class SyndesisHttpConfiguration {
         /**
          * Checks if we're using API connector-webhook.
          */
-        @ConditionalOnClass(name = "io.syndesis.connector.webhook.WebhookConnectorCustomizer")
+        @ConditionalOnResource(resources = "classpath:META-INF/syndesis/connector/webhook.json")
         static class WebhookUsed {
             // auto configuration test
         }

